### PR TITLE
Vol 141 plus ones

### PR DESCRIPTION
**Pull request by TobiasLounsbury**

### CODE REVIEW

File: CRM/Volunteer/Form/VolunteerSignUp.php [hunk 1]
```diff
@@ -232,38 +339,84 @@ function postProcess() {
     $values = $this->controller->exportValues();
 
     $profileFields = array();
-    foreach ($this->getProfileIDs() as $profileID) {
+    foreach ($this->getPrimaryVolunteerProfileIDs() as $profileID) {
       $profileFields += CRM_Core_BAO_UFGroup::getFields($profileID);
     }
     $profileValues = array_intersect_key($values, $profileFields);
     $activityValues = array_diff_key($values, $profileValues);
```
Comment (GinkgoFJG) [inline CRM/Volunteer/Form/VolunteerSignUp.php]: @TobiasLounsbury: my IDE is showing that $activityValues is unused. Looks like stuff got split up in refactoring. Should this (or $profileValues) be an argument to your new method createVolunteerActivity()?

File: CRM/Volunteer/Form/VolunteerSignUp.php [hunk 2]
```diff
@@ -232,38 +339,84 @@ function postProcess() {
...
-    // Search for duplicate
-    if (!$cid) {
-      $dedupeParams = CRM_Dedupe_Finder::formatParams($profileValues, 'Individual');
-      $dedupeParams['check_permission'] = FALSE;
-      $ids = CRM_Dedupe_Finder::dupesByParams($dedupeParams, 'Individual');
-      if ($ids) {
-        $cid = $ids[0];
-      }
+    $this->_primary_volunteer_id = $this->processProfileData($profileValues, $profileFields, $cid);
+    $activity_statuses = CRM_Activity_BAO_Activity::buildOptions('status_id', 'create');
```
Comment (GinkgoFJG) [inline CRM/Volunteer/Form/VolunteerSignUp.php]: @TobiasLounsbury: Looks like this belongs in createVolunteerActivity(), too.

Comment (TobiasLounsbury) [inline CRM/Volunteer/Form/VolunteerSignUp.php]: Yes, indeed.

File: CRM/Volunteer/Form/VolunteerSignUp.php (hunk 1 above)
Comment (TobiasLounsbury) [inline CRM/Volunteer/Form/VolunteerSignUp.php]: Your IDE is correct, and I've adapted the function to take the two missing vars as params

**Pull request merged**